instance/serializers/openedx_instance.py

```diff
@@ -75,8 +75,17 @@ def to_representation(self, obj):
             output['is_healthy'] = None
             output['is_steady'] = None
 
-        newest_appserver = obj.appserver_set.order_by('-created').first()
-        output['newest_appserver'] = AppServerBasicSerializer(newest_appserver, context=self.context).data
+        try:
+            # Note that appservers are ordered by '-created' by default.
+            # We don't change or check the ordering of the .appserver_set.all() queryset here
+            # because that causes the django ORM to force a new database query to be made
+            # for each instance here, even if appserver_set was previously cached.
+            assert OpenEdXAppServer._meta.ordering[0] == '-created'
+            newest_appserver = obj.appserver_set.all()[0]
```

Comment:
Any reason not to use `first()` here?

Edit: I see, it's probably to get an error if it does not exist.  Fair enough.  (You could also use `first()` and check the result.)